contents/javascript/VueJs/Intro.md

```diff
@@ -0,0 +1,156 @@
+# VueJs
+
+Author: [Lu Lechuan](https://github.com/LuLechuan)
+
+VueJs (also known as Vue) is an open-source [JavaScript framework](https://en.wikipedia.org/wiki/JavaScript_framework) for building user interfaces. It is designed to improve code quality and maintainablity.
+
+## Creating a Simple Project in VueJs
+
+#### Installation
+
+There are [multiple ways to install VueJs](https://vuejs.org/v2/guide/installation.html) and they are all easy. For example, one of the easiest ways is to run the npm command `$ npm install vue`.
+
+Apart from installing VueJs, you can install [VueJs development tools](https://github.com/vuejs/vue-devtools#vue-devtools) in your browser. This will allow you to inspect and debug the components that use VueJs in your projects.
+
+#### HelloWorld in VueJs
+
+This is a minimal example to show how easy it is to integrate VueJs into your web project:<br/><br/>
+The main HTML file
+```HTML
+<body>
+  <div id="root">
+    <h2>{{ message }}</h2>
+  </div>
+  <script src="https://unpkg.com/vue@2.5.13/dist/vue.js"></script>
+  <script src="the_path_to_the_javacript_file.js"></script>
+</body>
+```
+This is inside the javacript file
+```js
+new Vue ({
+  el: '#root',
+
+  data: {
+    message: "Hello World"
+  }
+});
+```
+
+Step-by-step explanation of the code:
+
+<b>Step 1:</b> Import VueJs CDN and the JavaScript file in the main HTML file.
+```HTML
+  <script src="https://unpkg.com/vue@2.5.13/dist/vue.js"></script>
+  <script src="the_path_to_the_javacript_file.js"></script>
+```
+
+<b>Step 2:</b> Create an instance of Vue (Vue is an object) in the JavaScript file; bind the instance to one of the component in our html file (e.g. create a component with id `root` and bind it with the instance of Vue).<br/>
+In this case, only the `root` component can be accessed in VueJs while the rest are unaffected. This is how we progressively plug in VueJs into our projects.
+
+```js
+  new Vue ({
+    el: '#root',
+  });
+```
+```HTML
+  <div id="root"></div>
+```
+
+<b>Step 3:</b> Specify our data (message: "Hello World") in the instance of Vue Class.
+```js
+  data: {
+    message: "Hello World"
+  }
+```
+
+<b>Step 4:</b> Pass the message to the HTML file using double curly brackets.
+```HTML
+  <div id="root">
+    <h2>{{ message }}</h2>
+  </div>
+```
+
+<b>Step 5:</b> Open the brower and we will see "Hello World" being displayed:
+> <h2>Hello World</h2>
+
+## Advantages and Disadvantages of VueJs
```

Comment:
This section still has the indentation problem?
![image](https://user-images.githubusercontent.com/1673303/38972343-78354080-43d2-11e8-8249-083a3873eaf5.png)